test/pod.bats

```diff
@@ -533,3 +533,25 @@ EOF
 
 	crictl run "$TESTDIR"/memory.json "$TESTDATA"/sandbox_config.json
 }
+
+@test "run container with default annotations" {
+	setup_crio
+
+	cat << EOF > "$CRIO_CONFIG_DIR/99-ann.conf"
+[crio.runtime]
+default_runtime = "ann"
+[crio.runtime.runtimes.ann]
+runtime_path = "$RUNTIME_BINARY_PATH"
+default_annotations = { "hello" = "1234", "pod" = "5678" }
+EOF
+	unset CONTAINER_DEFAULT_RUNTIME
+	unset CONTAINER_RUNTIMES
+
+	start_crio_no_setup
+
+	ctr_id=$(crictl run "$TESTDATA"/container_sleep.json "$TESTDATA"/sandbox_config.json)
+	annotations=$(crictl inspect "$ctr_id" | jq .info.runtimeSpec.annotations)
+	grep hello <<< "$annotations"
+	# pod spec should override default annotations
+	grep -v "5678" <<< "$annotations"
+}
```

Comment:
You need to exclude this test or make it work in kata:

```
not ok 135 run container with default annotations
# (in test file pod.bats, line 564)
#   `ctr_id=$(crictl run "$TESTDATA"/container_sleep.json "$TESTDATA"/sandbox_config.json)' failed
# time="2024-12-18T10:47:50.090479464Z" level=info msg="Updating config from single file: /etc/crio/crio.conf"
# time="2024-12-18T10:47:50.090505488Z" level=info msg="Updating config from drop-in file: /etc/crio/crio.conf"
# time="2024-12-18T10:47:50.0913983Z" level=info msg="Updating config from path: /etc/crio/crio.conf.d"
# time="2024-12-18T10:47:50.091518626Z" level=info msg="Updating config from drop-in file: /etc/crio/crio.conf.d/01-ns-lifecycle.conf"
# time="2024-12-18T10:47:50.091604066Z" level=info msg="Updating config from drop-in file: /etc/crio/crio.conf.d/01-overlay.conf"
# time="2024-12-18T10:47:50.091644374Z" level=info msg="Updating config from drop-in file: /etc/crio/crio.conf.d/99-log-level.conf"
# level=info msg="Using default capabilities: CAP_CHOWN, CAP_DAC_OVERRIDE, CAP_FSETID, CAP_FOWNER, CAP_SETGID, CAP_SETUID, CAP_SETPCAP, CAP_NET_BIND_SERVICE, CAP_KILL"
# time="2024-12-18T10:47:50.129665841Z" level=info msg="Updating config from path: "
# level=info msg="Using default capabilities: CAP_CHOWN, CAP_DAC_OVERRIDE, CAP_FSETID, CAP_FOWNER, CAP_SETGID, CAP_SETUID, CAP_SETPCAP, CAP_NET_BIND_SERVICE, CAP_KILL"
# time="2024-12-18T10:47:50Z" level=fatal msg="validate service connection: validate CRI v1 runtime API for endpoint \"unix:///tmp/tmp.lvCNqQ86CV/crio.sock\": rpc error: code = Unavailable desc = connection error: desc = \"transport: Error while dialing: dial unix /tmp/tmp.lvCNqQ86CV/crio.sock: connect: no such file or directory\""
# E1218 10:47:51.470463   62497 log.go:32] "CreateContainer in sandbox from runtime service failed" err=<
# 	rpc error: code = Unknown desc = container create failed: flag provided but not defined: -systemd-cgroup
# 	Usage of /opt/kata/bin/containerd-shim-kata-v2:
# 	  -address string
# 	    	grpc address back to main containerd
# 	  -bundle string
# 	    	path to the bundle if not workdir
# 	  -debug
# 	    	enable debug output in logs
# 	  -id string
# 	    	id of the task
# 	  -namespace string
# 	    	namespace that owns the shim
# 	  -publish-binary string
# 	    	path to publish binary (used for publishing events), but /opt/kata/bin/containerd-shim-kata-v2 will ignore this flag, please use the TTRPC_ADDRESS env
# 	  -socket string
# 	    	socket path to serve
# 	  -v	show the shim version and exit
#  > podSandboxID="361574031e833c28e96e692b540c180d83c3c0090173c17c158de79f8fade5dc"
# time="2024-12-18T10:47:51Z" level=fatal msg="running container: creating container failed: rpc error: code = Unknown desc = container create failed: flag provided but not defined: -systemd-cgroup\nUsage of /opt/kata/bin/containerd-shim-kata-v2:\n  -address string\n    \tgrpc address back to main containerd\n  -bundle string\n    \tpath to the bundle if not workdir\n  -debug\n    \tenable debug output in logs\n  -id string\n    \tid of the task\n  -namespace string\n    \tnamespace that owns the shim\n  -publish-binary string\n    \tpath to publish binary (used for publishing events), but /opt/kata/bin/containerd-shim-kata-v2 will ignore this flag, please use the TTRPC_ADDRESS env\n  -socket string\n    \tsocket path to serve\n  -v\tshow the shim version and exit\n"
```

https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/test-platform-results/pr-logs/pull/cri-o_cri-o/8829/pull-ci-cri-o-cri-o-main-ci-fedora-kata/1869327673696718848/artifacts/fedora-kata/cri-o-gather/artifacts/testout.txt